Framework/PythonInterface/plugins/algorithms/OrderWorkspaceHistory.py

```diff
@@ -67,38 +68,45 @@ def write_ordered_workspace_history(self):
 
         self.log().debug("Found {0} history files".format(len(historyfiles)))
 
-        # Read each history in as a list of tuples
-        original_lines = set()
-        for infile in historyfiles:
-            with open(os.path.join(source_folder, infile), 'r') as f:
-                for line in f:
-                    original_lines.add(line)
-
-        commands = []
-        # Remove any comment lines
-        for line in original_lines:
-            stripped_line = line.lstrip()
-            if stripped_line[0] is not '#':
-                commands.append(stripped_line)
-
-        self.log().debug("Found {0} unique commands".format(len(commands)))
-
-        # Add lists of histories together
-        all_unique_commands = [command.strip().split('#') for command in commands]
-
-        # Convert the datetime into a sortable integer
-        all_unique_commands = [(i[0], self._concatenate_iso_datetime(i[1]))
-                               for i in all_unique_commands]
+        all_lines = []
+
+        for fn in historyfiles:
```

Comment:
The way this list is generated can be replaced by a single call to [`glob.iglob`](https://docs.python.org/2/library/glob.html#glob.iglob).

e.g.
```python
for fn in glob.iglob(os.path.join(self.getPropertyValue(_recovery_folder), '*.py')):
    with open(fn) as f:
        # do stuff here
```